CRM/Price/BAO/PriceSet.php

```diff
@@ -1681,17 +1661,17 @@ protected static function addPriceFieldsToForm(CRM_Core_Form $form, $feeBlock, b
     $hideAdminValues = !CRM_Core_Permission::check('edit contributions');
     // CRM-14492 Admin price fields should show up on event registration if user has 'administer CiviCRM' permissions
     $adminFieldVisible = CRM_Core_Permission::check('administer CiviCRM');
+    $checklifetime = FALSE;
     foreach ($feeBlock as $id => $field) {
       if (CRM_Utils_Array::value('visibility', $field) == 'public' ||
         (CRM_Utils_Array::value('visibility', $field) == 'admin' && $adminFieldVisible == TRUE) ||
         !$validFieldsOnly
       ) {
         $options = $field['options'] ?? NULL;
         if ($className == 'CRM_Contribute_Form_Contribution_Main' && $component = 'membership') {
-          $userid = $form->getVar('_membershipContactID');
-          $checklifetime = self::checkCurrentMembership($options, $userid);
-          if ($checklifetime) {
-            $form->assign('ispricelifetime', TRUE);
+          $contactId = $form->getVar('_membershipContactID');
+          if ($contactId && $options) {
+            $checklifetime ?: self::checkCurrentMembership($options, $contactId);
```

Comment:
does that work - I thought it would need to be  

```
 $checklifetime = $checklifetime ?: self::checkCurrentMembership($options, $contactId);
```

if it does work - does it work on all php versions

Reply:
No, you're right - at the end of the night I was looking at small ways to improve my code, meaning a) it was after I'd done testing, b) it was late.  This was a mistake.